test/Controller/Plugin/ForwardTest.php

```diff
@@ -237,6 +238,35 @@ function ($e) {}
         $this->assertEquals(['content' => 'ZendTest\Mvc\Controller\TestAsset\ForwardController::testAction'], $result);
     }
 
+    public function testProblemListenersAreDetachedAndReattachedWhenPluginDispatchsRequestedController()
```

Comment:
@froschdesign I'm not sure what the ZF policy or practice is for this type of assertion but in this case, the assertion is implied in the mock method invocation expectation:
```php
$sharedEvents->expects($this->once())->method('detach')
```

If the method isn't invoked with the expected arguments, the test will fail with an exception. What is the preferred way of testing something like this? Maybe a test spy that monitors called methods and then the assertion can be the method was called?